In [1]:
mu <- .8893
# Major => 1 = Econ, 2 = Anthro, 3 = Sociology
a1 <- 0
a2 <- 1.9860
a3 <- 1.1889
# High School Back Ground (BG) => 1 = Rural, 2 = Urban
eta1 <- 0
eta2 <- 1.2564
# Interaction
gam11 <- 0
gam12 <- 0
gam21 <- 0
gam22 <- -1.6631
gam31 <- 0
gam32 <- -1.6130

```R
                                  Estimate  Std. Error  t value  Pr(>|t|)
(Intercept)                         0.8893      0.4033    2.205   0.03554 *
as.factor(Major)2                   1.9860      0.6377    3.114   0.00413 **
as.factor(Major)3                   1.1889      0.6377    1.864   0.07244 .
as.factor(BG)2                      1.2564      0.5207    2.413   0.02237 *
as.factor(Major)2:as.factor(BG)2   -1.6631      0.8233   -2.020   0.05270 .
as.factor(Major)3:as.factor(BG)2   -1.6130      0.8233   -1.959   0.05977 .
```

In [2]:
(E_y11 <- mu + a1 + eta1 + gam11) # = mu
(E_y12 <- mu + a1 + eta2 + gam12) # = mu + eta2
(E_y21 <- mu + a2 + eta1 + gam21) # = mu + a2
(E_y22 <- mu + a2 + eta2 + gam22) # = mu + a2 + eta2 + gam22
(E_y31 <- mu + a3 + eta1 + gam31) # = mu + a3
(E_y32 <- mu + a3 + eta2 + gam32) # = mu + a3 + eta2


[1] 0.8893

[1] 2.1457

[1] 2.8753

[1] 2.4686

[1] 2.0782

[1] 1.7216

### 1a) Which group of students has the lowest score? (What is it?) Which group of students has the highest score? (What is it?)

Note that 

$$
\begin{array}{rclclcr}
    E[y_{11}] &=& \mu + \alpha_1 + \eta_1 + \gamma_{11} &\approx& \hat\mu &=& .8893 \\
    E[y_{12}] &=& \mu + \alpha_1 + \eta_2 + \gamma_{12} &\approx& \hat\mu +\hat\eta_2 &=& 2.1457 \\
    E[y_{21}] &=& \mu + \alpha_2 + \eta_1 + \gamma_{21} &\approx& \hat\mu +\hat\alpha_2 &=& 2.8753 \\
    E[y_{22}] &=& \mu + \alpha_2 + \eta_2 + \gamma_{22} &\approx& \hat\mu + \hat\alpha_2 + \hat\eta_2 + \hat\gamma_{22}&=& 2.4686 \\
    E[y_{31}] &=& \mu + \alpha_3 + \eta_1 + \gamma_{31} &\approx& \hat\mu +\hat\alpha_3 &=& 2.0782 \\
    E[y_{32}] &=& \mu + \alpha_3 + \eta_2 + \gamma_{32} &\approx& \hat\mu + \hat\alpha_3 + \hat\eta_2 + \hat\gamma_{32}&=& 1.7216 \\
\end{array}
$$


So, the group that corresponds to $y_{11}$ has the **lowest** math ineptitude score. That is, **Economics** majors from **rural** high school backgrounds, with a score of **.8893**.

The group that corresponds to $y_{21}$ has the **highest** math ineptitude score. That is, **Anthropology** majors from **rural** high school backgrounds, with a score of **2.8753**.


```R
Residual standard error: 0.988 on 29 degrees of freedom
Multiple R-squared: 0.3057, Adjusted R-squared: 0.1859
F-statistic: 2.553 on 5 and 29 DF, p-value: 0.04945
```

### 1b) In the `summary(.)` output there is an F-statistic, F= 2:553 with 5 and 29 degrees offreedom.

1. What are the null and alternative hypotheses being tested?
2. What conclusion would you make? (Please state in general terms that relate to the groups rather than parameters).

$H_0: \alpha_1 = \alpha_2 = \alpha_3 = \eta_1 = \eta_2 = 0$

$H_1$: At least one of the parameters is not 0.

Heuristically, the null hypothesis is that the effect of majors and high school background is not significantly different from zero. The alternative hypothesis is that at least one major or background significantly different from zero and, hence, significantly effects math ineptitude scores (at the 95% confidence level).


### 1c) In the `anova(.)` output the p-value on the line corresponding to BG is large, yet in the summary from `lm` the p-value for BG2 is small. Do the p-values from the two summaries contradict each other? Explain what is being tested and what it means in this context. Is the students background relevant for predicting the score?

```R
> anova(lm(Score1 ~ as.factor(Major) * as.factor(BG), data=dat))  

  Analysis of Variance Table  
  Response: Score1
                                 Df  Sum Sq  Mean Sq  F value  Pr(>F)  
  as.factor(Major)                2  6.0755  3.03776  3.1123  0.05964 .  
  as.factor(BG)                   1  0.8623  0.86233  0.8835  0.35502  
  as.factor(Major):as.factor(BG)  2  5.5228  2.76142  2.8291  0.07543 .  
  Residuals 29 28.3058 0.97606  
```

In the ANOVA table, the sum of squares corresponding to BG is the reduction in sum of squares by introducing High School Background as a parameter in the model **after** having introduced college Major (and a grand mean or intercept) as parameters to the model. Similarly, the p-value corresponding to BG is the p-value associated with the null hypothesis that high school background (BG) does not improve the model with the intercept and college major as parameters. In other words, in the ANOVA table, the hypothesis $H_0: \eta_1 = \eta_2 = 0$. This means that sequentially added to the model, high school background does not contribute to a reduction in the sums of squares at the 95% confidence level.

In the summary table, the p-value corresponding to BG2 is smaller than that of BG in ANOVA (in fact, it's smaller than .05). The hypothesis being tested is not the same. In this case, the hypothesis is $H_0: \eta_2 = 0$. $\eta_2$ is not being added sequentially to the model. Rather all parameters are fit in the model simultaneously. In this case, high school background contributes significantly to the model in predicting mathematics ineptitude.

For completeness, the two fits (in `anova` and `lm`) *should* be the same, assuming the data was not tampered with to trick us... But the *reporting* is different.

$
%http://stats.stackexchange.com/questions/28938/why-do-linear-regression-and-anova-give-different-p-value-in-case-of-consideri
$

### 2a) Estimate $\beta_1,\beta_2,$ and $\sigma^2$

In [3]:
y <- c(82, 79, 74, 83, 80, 81, 84, 81)
x1 <- c(10, 9, 9, 11, 11, 10, 10, 12)
x2 <- c(15, 14, 13, 15, 14, 14, 16, 13)
n <- length(y)
X <- cbind(1,x1,x2)
Xt <- t(X)
XtX <- Xt %*% X
beta.hat <- solve(XtX,Xt%*%y)
s2.hat <- sum((X%*%beta.hat-y)^2)  / (n-3)
rbind(beta.hat,s2.hat)

[,1]
       30.000
x1      1.625
x2      2.375
s2.hat  1.550

$\begin{pmatrix}
\hat\beta_1 \\
\hat\beta_2 \\
\hat\sigma^2 \\
\end{pmatrix}$ =
$\begin{pmatrix}
1.625\\
2.375\\
1.550\\
\end{pmatrix}$

In [4]:
#summary(lm(y ~ x1 + x2)) # check if same

### 2b) Give a 95% confidence interval for $\beta_1$ and $\beta_1 + \beta_2$

In [5]:
C <- matrix(c(0,1,0,
              0,1,1),2,byrow=TRUE)
Cb <- C %*% beta.hat
se <- s2.hat * C %*% solve(XtX) %*% t(C)
t.star <- qt(.975,n-3)

b1_CI <- Cb[1] + c(-1,1) * t.star * sqrt( se[1,1] )
b1_plus_b2_CI <- Cb[2] + c(-1,1) * t.star * sqrt( se[2,2] )

In [10]:
rbind(b1_CI, b1_plus_b2_CI)

[,1]     [,2]
b1_CI         0.4537923 2.796208
b1_plus_b2_CI 2.2893417 5.710658

The 95% confidence interval for $\beta_1$ is **(.4538, .2796)**.

The 95% confidence interval for $\beta_1+\beta_2$ is **(2.289, 5.711)**.

### 2c)  Perform an $\alpha = .01$ test for $H_0: \beta_2 = 3$

In [15]:
b2 <- beta.hat[2]
se.b2 <- sqrt( s2.hat * solve(XtX)[3,3] )
b2_CI <- b2 + c(-1,1) * qt(.995,n-3) * se.b2
rbind(b2_CI)

[,1]     [,2]
b2_CI -0.2121237 3.462124

The 95% confidence interval for $\beta_2$ is **(-.2121, 3.462)**.

Since the confidence interval contains 3, at the 95% confidence level, we **fail to reject** the null hypothesis that $\beta_2 = 3$.

### 2d) Find the p-value for the test of $H_0: \beta_1 - \beta_2 = 0$.

In [19]:
lam <- matrix(c(0,1,-1),nrow=1)
b1_minus_b2 <- lam %*% beta.hat 
se_d <- sqrt( s2.hat * lam %*% solve(XtX) %*% t(lam) )

t.stat <- b1_minus_b2 / se_d
p.val <- pt(abs(t.stat),n-3,lower.tail=FALSE) * 2

t.stat
p.val

[,1]
[1,] -1.204829

[,1]
[1,] 0.2821799

The p-value is .28. So, at the 95% confidence level, we **fail to reject** the null hypothesis that the difference between $\beta_1$ and $\beta_2$ is 0.

### 3) Show that for a linear model with an intercept, $R^2$ is simply the square of the correlation between the data $y_i$ and the predicted values $\hat{y}_i$.

$$
\def\y{\boldsymbol y}
\def\P{\boldsymbol P}
\def\I{\boldsymbol I}
\def\ds{\displaystyle}
\def\bk#1{\left[#1\right]}
\def\p#1{\left(#1\right)}
$$

We need to show that $R^2 = \ds\frac{\y^T(\P_x-\P_1)\y}{\y^T(\I-\P_1)\y} = 
\p{\frac{(\y-\bar\y)^T(\hat\y-\bar\y)}{\ds\sqrt{(\y-\bar\y)^T(\y-\bar\y)(\hat\y-\bar\y)^T(\hat\y-\bar\y)}}}^2 =
\p{\text{Cor}(\y,\hat{\y})}^2$.

First, note that by adding and subtracting $\hat\y$,

$$
\begin{array}{rcl}
    (\y-\bar\y)^T(\hat\y-\bar\y) &=& \bk{(\hat\y - \bar\y) + (\y-\hat\y)}^T(\hat\y-\bar\y) \\
    &=& (\hat\y - \bar\y)^T(\hat\y-\bar\y) + (\y-\hat\y)^T(\hat\y-\bar\y) \\
    &=& (\hat\y - \bar\y)^T(\hat\y-\bar\y) + 0.
\end{array}
$$

Then, Cor($\y,\hat\y$) can be computed as follows:

$$
\begin{array}{rcl}
    \text{Cor}(\y,\hat\y) &=& \frac{\ds(\y-\bar\y)^T(\hat\y-\bar\y)}{\ds\sqrt{(\y-\bar\y)^T(\y-\bar\y)(\hat\y-\bar\y)^T(\hat\y-\bar\y)}} \\
    &=& \frac{\ds(\hat\y - \bar\y)^T(\hat\y-\bar\y)}{\ds\sqrt{(\y-\bar\y)^T(\y-\bar\y)(\hat\y-\bar\y)^T(\hat\y-\bar\y)}} \\
    &=& \sqrt\frac{\ds{(\hat\y - \bar\y)^T(\hat\y-\bar\y)}}{\ds{(\y-\bar\y)^T(\y-\bar\y)}} \\
    &=& \sqrt\frac{\ds{\hat\y^T\hat\y - n\bar{y}^2}}{\ds{\y^T\y-n\bar{y}^2}} \\
\end{array}
$$

Now, 
$$
\begin{array}{rcl}
    R^2 &=& \ds\frac{\y^T(\P_x-\P_1)\y}{\y^T(\I-\P_1)\y} \\
    &=& \ds\frac{\y^T\P_x\y - \y^T\P_1\y}{\y^T\y - \y^T\P_1\y}
    &=& \ds\frac{\hat\y^T\hat\y - \bar\y^T\bar\y}{\y^T\y - \bar\y^T\bar\y} \quad(\because \text{ projection matrices are symmetric and idempotent}) \\
    &=& \ds\frac{\hat\y^T\hat\y - n\bar{y}^2}{\y^T\y - n\bar{y}^2}
\end{array}
$$

Therefore, $R^2 = \p{\text{Cor}(\y,\hat\y)}^2$.